pkg/operator/ceph/csi/controller.go

```diff
@@ -295,29 +277,14 @@ func (r *ReconcileCSI) reconcile(request reconcile.Request) (reconcile.Result, e
 		}
 		clusterInfo.OwnerInfo = k8sutil.NewOwnerInfo(&cephClusters.Items[i], r.scheme)
 
-		// is holder enabled for this cluster?
-		thisHolderEnabled := (!csiHostNetworkEnabled || cluster.Spec.Network.IsMultus()) && !csiDisableHolders
-
-		// Do we have a multus cluster or csi host network disabled?
-		// If so deploy the plugin holder with the fsid attached
-		if thisHolderEnabled {
-			logger.Debugf("cluster %q: deploying the ceph-csi plugin holder", cluster.Name)
-			r.clustersWithHolder = append(r.clustersWithHolder, ClusterDetail{cluster: &cephClusters.Items[i], clusterInfo: clusterInfo})
-
-			// holder pods are enabled globally if any cluster needs a holder pod
-			holderEnabled = true
-		} else {
-			logger.Debugf("cluster %q: not deploying the ceph-csi plugin holder", request.NamespacedName)
-		}
-
-		// if holder pods were disabled, the controller needs to update the configmap for each
-		// cephcluster to remove the net namespace file path
+		// ensure any remaining holder-related configs are cleared
+		holderEnabled = false
 		err = reconcileSaveCSIDriverOptions(r.context.Clientset, cluster.Namespace, clusterInfo)
 		if err != nil {
 			return opcontroller.ImmediateRetryResult, errors.Wrapf(err, "failed to update CSI driver options for cluster %q", cluster.Name)
 		}
 
-		// Skip the new CSI-operator creation when holder pod is enabled until multus support is added in the CSI operator
+		// disable Rook-managed CSI drivers if CSI operator is enabled
 		if EnableCSIOperator() {
```

Comment:
With the slight change to `EnableCSIOperator()` [noted here](https://github.com/rook/rook/pull/14819/files#r1813801700) I think the comment here changes slightly. @subhamkrai make sure this looks good.

Reply:
the condition looks good to me.